stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -138,15 +139,41 @@ internal class PaymentOptionsActivity : BasePaymentSheetActivity<PaymentOptionRe
                 }
             }
         }
+    }
+
+    private val label: String
+        get() = resources.getString(
+            R.string.stripe_paymentsheet_add_button_label
+        )
 
-        viewModel.userSelection.observe(this) { paymentSelection ->
-            if (paymentSelection != null) {
-                onActionCompleted(paymentSelection)
+    @VisibleForTesting
+    private val viewStateObserver:  (ViewState.Add?, DefaultPrimaryButton) -> Unit
+        get() = { state, addButton ->
+            addButton.updateState(state)
```

Comment:
Do we need `DefaultPrimaryButton` as an argument? We can just do `viewBinding.addButton`.